src/app/core/views/Preferences/tabs/Billing/CancelSubscriptionModal.tsx

```diff
@@ -24,30 +28,60 @@ const CancelSubscriptionModal = ({
   cancelSubscription: (feedback: string) => void;
 }): JSX.Element => {
   const { translate } = useTranslationContext();
-  const [step, setStep] = useState<1 | 2>(1);
+  const [step, setStep] = useState<1 | 2 | 3>(2);
+  const [couponAvailable, setCouponAvailable] = useState(false);
+  const dispatch = useAppDispatch();
 
   useEffect(() => {
-    if (isOpen) {
-      setStep(1);
+    paymentService.getCoupon().then((coupon) => setCouponAvailable(coupon.elegible));
+  }, []);
+
+  useEffect(() => {
+    couponAvailable && setStep(1);
+  }, [couponAvailable]);
+
+  const applyCoupon = async () => {
+    try {
+      await paymentService.applyCoupon();
+      notificationsService.show({ text: translate('notificationMessages.successApplyCoupon') });
+    } catch (error) {
+      console.error(error);
+      notificationsService.show({
+        text: translate('notificationMessages.errorApplyCoupon'),
+        type: ToastType.Error,
+      });
+    } finally {
+      onClose();
+      setTimeout(() => {
```

Comment:
Why? It this for refreshing the UI? Then, we should just refresh the information in the case that an error does not happen, isn't it?

Reply:
It's for this

Reply:
Done